cmd/envtool/envtool.go

```diff
@@ -98,30 +97,8 @@ func setupAnyPostgres(ctx context.Context, logger *zap.SugaredLogger, uri string
 		return err
 	}
 
-	// TODO https://github.com/FerretDB/FerretDB/issues/3452
+	
 
-	var pgPool *pgdb.Pool
-
-	var retry int64
-	for ctx.Err() == nil {
-		if pgPool, err = pgdb.NewPool(ctx, uri, logger.Desugar(), sp); err == nil {
-			break
-		}
-
-		logger.Infof("%s: %s", uri, err)
-
-		retry++
-		ctxutil.SleepWithJitter(ctx, time.Second, retry)
-	}
-
-	if ctx.Err() != nil {
-		return ctx.Err()
-	}
-
-	defer pgPool.Close()
-
-	return nil
-}
```

Comment:
That code waited for PostgreSQL to start and checked connection and authentication. We still need to do that, but by using new backend code.

Reply:
Thank you for quick response @AlekSi  & for the explanation 
I see in the issue you stated that we probably need to use `metadata/pool` package
Can give more Information on that?
And lastly I'm supposed to work on only that section of the code you mentioned?

Reply:
> Can you give more Information on that?

https://github.com/FerretDB/FerretDB/blob/main/internal/backends/postgresql/metadata/pool/pool.go
Create a new pool, call `Pool.Get` until it stops returning an error, close the pool.

> And lastly I'm supposed to work on only that section of the code you mentioned?

Yes, we just need to replace that part of the code